lib/core/Stage.cpp

```diff
@@ -135,10 +138,34 @@ Stage::~Stage() {
     stop_thread = true;
     if (this_thread.joinable())
         this_thread.join();
+    unregister_tid();
 }
 
 std::string Stage::dot_string(const std::string& prefix) const {
     return fmt::format("{:s}\"{:s}\" [shape=box, color=darkgreen];\n", prefix, get_unique_name());
 }
 
+// Used to get tid from pthread_t
+struct pthread_fake {
+    char offset[720];
+    pid_t tid;
+    void* others;
+};
+
+void Stage::register_tid(pthread_t ptr) {
+    pid_t tid = ((pthread_fake*)ptr)->tid;
+    thread_list[unique_name] = tid;
+}
+
+void Stage::unregister_tid() {
```

Comment:
As a follow up to the comment above, this should likely take a `ptr` convert to a `tid` and remove it from the vector.